### PR TITLE
Small refactor to improve readability

### PR DESCRIPTION
**Pull request by nicl**
Tweaks some namings and replaces RepoAndOwner class with an interface. The main aim is to make the handler more readable - some of this is subjective for sure but overall I think this is an improvement.

### CODE REVIEW

File: packages/repo-fetcher/src/handler.ts [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
+		const adminRepos: string[] = getAdminReposFromResponse(teamRepos);
+		return adminRepos.map((repoName) => ({
+			teamSlug: team.slug,
+			repoName,
+		}));
+	});
+
+	const flattened = (await Promise.all(teamRepositories)).flat();
+
+	return flattened.reduce<Record<string, string[] | undefined>>((acc, repo) => {
```
Comment (akash1810) [inline packages/repo-fetcher/src/handler.ts:30]: Would `Record<string, string[]>` be an easier type than `Record<string, string[] | undefined` here? Where the empty array is the `undefined` case.

Comment (nicl) [reply packages/repo-fetcher/src/handler.ts:30]: The problem is, we can't guarantee what is returned will always return a `string[]` for a given key (lookup) string.

Typescript will happily let you declare:

`Record<string, string[]>`

but really it's just an object, and there's no guarantee that x[foo] will return anything (as `foo` may not exist in `x`).

Comment (akash1810) [reply packages/repo-fetcher/src/handler.ts:30]: True, however we're currently using nullish coalescing to handle that case, I think?

**Review by akash1810 — APPROVED** [review]
Added some comments, none of which are blocking.

File: packages/repo-fetcher/src/handler.ts (hunk 1 above)
Comment (nicl) [reply packages/repo-fetcher/src/handler.ts:30]: Ah to be clear, the value when a string is present will always be string[] but the `undefined` case is what is returned when you lookup a string in the record that doesn't exist.

**Pull request merged**